src/server/replica.cc

```diff
@@ -457,7 +458,8 @@ error_code Replica::InitiateDflySync() {
   RETURN_ON_ERR(cntx_.SwitchErrorHandler(std::move(err_handler)));
 
   // Make sure we're in LOADING state.
-  CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING) == GlobalState::LOADING);
+  CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING).first ==
```

Comment:
I don't think it's applicable to the current situation. Because we still need to provide an expected state

Reply:
what do you mean? The expected state is always ACTIVE in these cases.

Reply:
i.e. maybe the name should be `SwitchFromActive`

Reply:
I get it now, we can create such a function, but we get incorrect behavior because the existing function returns the current state when expected is wrong and the current state can have been already in the target state)
I think you want to suggest SwitchFromActiveOrCheck

Reply:
maybe the original function does something else but the code here in all other places wants to switch from active :)
I am just saying that maybe the original API that I wrote is too cumbersome. anyway, it does not have to be part of this PR, I can check it later